### PR TITLE
Feat39 clocking change

### PR DESCRIPTION
**Pull request by Enyuuus**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] My code runs
- [x] My code follows the coding guidelines
- [x] I rebased my branch on development
- [x] My code only has comments when necessary for understandability 
- [x] I removed commented out code
- [x] My code corresponds with the documentation
- [x] I have updated the documentation when changes were needed
- [x] All unit-tests succeeded (Except for not implemented exceptions)

# Optional:

- [ ] Is the branch linked to an issue?

### CODE REVIEW

File: Bumbodium/Views/Clocking/AddClockingHour.cshtml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        <div class="row p-2 bg-white">
+            <div class="col d-flex justify-content-start">
+                <input type="hidden" asp-for="Date" />
+                <select asp-for="Name" asp-items="ViewBag.Options" required>
+                    <option selected="selected" value="" disabled>Selecteer</option>
+                </select>
+            </div>
+            <div class="col d-flex justify-content-start"><h5 class="m-2">Begin tijd :</h5><input type="time" asp-for="ClockStartTime" /></div>
+            <div class="col d-flex justify-content-start"><h5 class="m-2">Eind tijd :</h5><input type="time" asp-for="ClockEndTime" /></div>
+            <div class="col d-flex justify-content-start"><h5 class="m-2">werknemer is ziek :</h5><input type="checkbox" asp-for="IsSick" /></div>
```
Comment (Martijn1030) [inline Bumbodium/Views/Clocking/AddClockingHour.cshtml:18]: "Ziek:" is genoeg denk ik

File: Bumbodium/Views/Clocking/AddClockingHour.cshtml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            </div>
+            <div class="col d-flex justify-content-start">
+                <span asp-validation-for="ClockEndTime" class="text-danger"></span>
+            </div>
+            <div class="col-1 d-flex justify-content-start"></div>
+        </div>
+        <br />
+        <div class="row pb-0">
+            <div class="col text-center p-0 ">
+                <button type="submit" class="btn btn-lg btn-primary btn-outline-warning">Input aanpassen</button>
```
Comment (Martijn1030) [inline Bumbodium/Views/Clocking/AddClockingHour.cshtml:32]: Je past hier niks aan, dit is de add dus "Toevoegen" lijkt me logischer

File: Bumbodium/Views/Clocking/IndexManager.cshtml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+                    <input type="hidden" asp-for="ClockingDay[day.Key][i].Date" />
+                    <div class="row p-1 bg-white shadow">
+                        <div class="col d-flex justify-content-start"><h5 class="m-2">Begin tijd :</h5><input type="time" asp-for="ClockingDay[day.Key][i].ClockStartTime" /></div>
+                        <div class="col d-flex justify-content-start"><h5 class="m-2">Eind tijd :</h5><input type="time" asp-for="ClockingDay[day.Key][i].ClockEndTime" /></div>
+                        <div class="col d-flex justify-content-start"><h5 class="m-2">werknemer is ziek :</h5><input type="checkbox" asp-for="ClockingDay[day.Key][i].IsSick" /></div>
+                        <div class="col d-flex justify-content-start">
+                            <h5 class="m-2">Delete de uren :</h5>
+                            <a asp-action="DeleteClocking" asp-route-date="@Model.ClockingDay[day.Key][i].Date.Date.Ticks"
+                   asp-route-presenceId="@Model.ClockingDay[day.Key][i].PresenceId" class="m-auto"><span class="material-icons">delete</span></a>
```
Comment (Martijn1030) [inline Bumbodium/Views/Clocking/IndexManager.cshtml:45]: Tekst kan weg denk ik, verwijder icoontje is genoeg (mss rood maken)

File: Bumbodium/Models/ClockingView/ManagerClockingItem.cs [hunk 4]
```diff
@@ -1,12 +1,23 @@
...
         public DateTime? ScheduleStartTime { get; set; }
         public DateTime? ScheduleEndTime { get; set; }
+        public bool IsSick { get; set; }
+
+        public string Name { get; set; }
+
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            if (ClockStartTime >= ClockEndTime)
+                yield return new ValidationResult("Start tijd kan niet groter zijn dan eind tijd of gelijk aan elkaar.", new[] { nameof(ClockStartTime) });
```
Comment (Martijn1030) [inline Bumbodium/Models/ClockingView/ManagerClockingItem.cs:20]: "Start tijd kan niet na of gelijk aan eind tijd zijn" lijkt mij duidelijker hier

File: Bumbodium/Views/Clocking/IndexManager.cshtml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                            </div>
+                        </div>
+                    }
+                }
+                <br />
+            }
+        </div>
+        <br />
+        <div>
+            <input type="submit" class="btn btn-lg btn-primary btn-outline-warning m-3" value="Veranderen klok uren opslaan">
```
Comment (Martijn1030) [inline Bumbodium/Views/Clocking/IndexManager.cshtml:84]: Veranderen klok uren opslaan klinkt apart imo, "Aanpassingen opslaan" lijkt mij beter

**Review by Martijn1030 — CHANGES_REQUESTED** [review]
Mooi man, wel hier en daar wat tekst die wat simpler kan zijn denk ik

**Review by Martijn1030 — APPROVED** [review]

**Review by Bewerkt — APPROVED** [review]

**Pull request merged**